src/Pagination.tsx

```diff
@@ -588,9 +588,11 @@ const Pagination: React.FC<PaginationProps> = (props) => {
         selectPrefixCls={selectPrefixCls}
         changeSize={showSizeChanger ? changePageSize : null}
         pageSize={pageSize}
-        pageSizeOptions={pageSizeOptions}
+        pageSizeOptions={(typeof showSizeChanger === 'object' && showSizeChanger?.options) || pageSizeOptions}
```

Comment:
`showSizeChanger` should be assigned to a default object value if `showSizeChanger={true}`. Then we don't need `typeof showSizeChanger` here.

Reply:
ping